frontend/src/metabase/visualizations/components/ChartSettingsWidgetPopover.tsx

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import TippyPopover from "metabase/components/Popover/TippyPopover";
+import ChartSettingsWidget from "./ChartSettingsWidget";
+
+import { PopoverRoot, PopoverTitle } from "./ChartSettingsWidgetPopover.styled";
+
+interface Widget {
+  id: string;
+  props: Record<string, unknown>;
+}
+
+interface ChartSettingsWidgetPopoverProps {
+  anchor: HTMLElement;
+  handleEndShowWidget: () => void;
+  widgets: [Widget];
+  currentWidget: Widget;
+  extraWidgetProps: [];
+}
+
+export const ChartSettingsWidgetPopover = ({
+  anchor,
+  handleEndShowWidget,
+  widgets,
+  currentWidget,
+  extraWidgetProps,
+}: ChartSettingsWidgetPopoverProps) => {
+  const popoverWidgets =
```

Comment:
nit: This might be a bit misleading as we would eventually get an array of a single widget or an empty array. I think you could tweak this a little bit and make it be either a single widget or undefined and rename the variable to
```suggestion
  const popoverWidget =
```

to communicate that this is a single widget. What do you think?